reV/SAM/econ.py

```diff
@@ -160,57 +160,79 @@ def _get_annual_energy(site, site_df, site_gids, cf_arr, inputs, calc_aey):
-            added.
+        profiles : np.ndarray
+            2D array (time, n_sites) of all capacity factor profiles for all
+            the requested sites.
         """
 
-        # get the system capacity
-        sys_cap = Economic._parse_sys_cap(site, inputs, site_df)
-
         # Retrieve the generation profile for single owner input
         with Outputs(cf_file) as cfh:
 
             # get the index location of the site in question
-            site_gids = list(cfh.meta['gid'])
-            isite = site_gids.index(site)
+            site_gids = list(cfh.get_meta_arr('gid'))
+            isites = [site_gids.index(s) for s in sites]
 
             # look for the cf_profile dataset
             if 'cf_profile' in cfh.datasets:
-                gen = cfh['cf_profile', :, isite] * sys_cap
+                profiles = cfh['cf_profile', :, isites]
             elif 'cf_profile-{}'.format(cf_year) in cfh.datasets:
-                gen = (cfh['cf_profile-{}'.format(cf_year), :, isite]
-                       * sys_cap)
+                profiles = cfh['cf_profile-{}'.format(cf_year), :, isites]
             elif 'cf_profile_{}'.format(cf_year) in cfh.datasets:
-                gen = (cfh['cf_profile_{}'.format(cf_year), :, isite]
-                       * sys_cap)
+                profiles = cfh['cf_profile_{}'.format(cf_year), :, isites]
             else:
-                raise KeyError('Could not find cf_profile values for '
-                               'SingleOwner. Available datasets: {}'
-                               .format(cfh.datasets))
-        # add to input dict
+                msg = ('Could not find cf_profile values for '
+                       'input to SingleOwner. Available datasets: {}'
+                       .format(cfh.datasets))
+                logger.error(msg)
+                raise KeyError(msg)
+
+        return profiles
+
+    @staticmethod
+    def _make_gen_profile(isite, site, profiles, site_df, inputs):
+        """Get the single-site generation time series and add to inputs dict.
+
+        Parameters
+        ----------
+        isite : int
+            Site index in the profiles array.
+        site : int
+            Site resource GID.
+        profiles : np.ndarray
+            2D array (time, n_sites) of all capacity factor profiles for all
+            the requested sites.
+        site_df : pd.DataFrame
+            Dataframe of site-specific input variables. Row index corresponds
+            to site number/gid (via df.loc not df.iloc), column labels are the
+            variable keys that will be passed forward as SAM parameters.
+        inputs : dict
+            Dictionary of SAM input parameters.
+
+        Returns
+        -------
+        inputs : dict
+            Dictionary of SAM input parameters with the generation profile
+            added.
+        """
+
+        sys_cap = Economic._parse_sys_cap(site, inputs, site_df)
+        gen = profiles[:, isite] * sys_cap
```

Comment:
why not set this directly to the dict, i.e. don't create the intermediate gen variable?

Reply:
because thats how it was before hahah. Will change!